qiskit/extensions/hamiltonian_gate.py

```diff
@@ -127,7 +131,25 @@ def validate_parameter(self, parameter):
 
 
 def hamiltonian(self, operator, time, qubits, label=None):
-    """Apply hamiltonian evolution to qubits."""
+    """Apply hamiltonian evolution to qubits.
+
+    This gate resolves to a :class:`.UnitaryGate` as :math:`U(t) = exp(-i t H)`,
+    which can be decomposed into basis gates if it is 2 qubits or less, or
+    simulated directly in Aer for more qubits.
+
+    Args:
+        operator (matrix or Operator): a hermitian operator.
+        time (float): time evolution parameter.
```

Comment:
This code also allows `ParameterExpression` as type (see eg. L91 and following which cover this case), could you add that type in the initializer and this method here? 🙂

Reply:
does that apply to `operator` or `time`? I can't easily figure out where `validate_parameter` is used and the docstring is not very clear.

Reply:
That applies to `time` 🙂